### ISSUE TITLE
DisableObservabilityContextCustomizer can't be switched off through properties

### ISSUE THREAD

**Issue by asibross** [spring-projects/spring-boot#35009]
The introduction of [`ObservabilityContextCustomizerFactory`](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/actuate/observability/ObservabilityContextCustomizerFactory.java) and `@AutoConfigureObservability` is breaking smoke/integration tests that were relying on certain beans to exists.
Basically any auto-configuration that uses `@ConditionalOnEnabledTracing` when creating beans will be missed in the test context.
The purpose of this issue is to request that this disabling metric/tracing functionality be controlled by a property.

CC: @jonatan-ivanov

**Comment by philwebb** [comment]
We discussed this on today's team call and we think we should consider it a bug that there's no way to opt-out of the customizer without an annotation.

We think one way to fix this would be to change `DisableObservabilityContextCustomizer` so that if there's no annotation it looks for a property. We have some [existing testing properties](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#appendix.application-properties.testing), so it seems reasonable to add more.

Perhaps `spring.test.observability.auto-configure` where `true` is the same as `@AutoConfigureObservability(true, true)` and `false` is the same as `@AutoConfigureObservability(false, false)`. If an annotation is on the test then it takes precedence.
